test/integration/common/StudioActions.kt

```diff
@@ -89,43 +121,18 @@ object StudioActions {
         return absoluteDestination.toPath()
     }
 
-    /// Wait `timeMillis` milliseconds, then wait for all recompositions to finish.
     suspend fun delayAndRecompose(composeRule: ComposeContentTestRule, timeMillis: Int = Delays.RECOMPOSE) {
         delay(timeMillis.toLong())
-        composeRule.waitForIdle()
-    }
-
-    suspend fun waitForConditionAndRecompose(
-        context: ComposeContentTestRule,
-        failMessage: String,
-        beforeRetry: (() -> Unit) = {},
-        successCondition: () -> Boolean
-    ) {
-        var success = false
-        val deadline = System.currentTimeMillis() + 10_000
-        while (!success && System.currentTimeMillis() < deadline) {
-            try {
-                if (successCondition()) {
-                    success = true
-                } else {
-                    beforeRetry()
-                    delayAndRecompose(context, 500)
-                }
-            } catch (e: Exception) {
-                LOGGER.error(e.stackTraceToString())
-            }
-        }
-
-        if (!successCondition()) {
-            fail(failMessage)
-        }
```

Comment:
`androidx.compose.ui.test.junit4.ComposeTestRule.waitUntil` provides much of this logic, and this is the approach we use going forward as it is more in line accepted testing practices for Compose.